File: app/src/main/java/com/sysu/edu/api/AuthorizationJar.java
package com.sysu.edu.api;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.NonNull;

public class AuthorizationJar {
    private final Context context;
    private final SharedPreferences authPreferences;
    private final SharedPreferences tokenPreferences;
    private final SharedPreferences privacyPreferences;
    private final CookieManager cookieManager;
    
    public AuthorizationJar(Context context) {
        this.context = context;
        cookieManager = new CookieManager(context);
        authPreferences = context.getSharedPreferences("authorization", Context.MODE_PRIVATE);
        tokenPreferences = context.getSharedPreferences("token", Context.MODE_PRIVATE);
        privacyPreferences = context.getSharedPreferences("privacy", Context.MODE_PRIVATE);
    }
    
    public String getAuthorization(String host) {
        return authPreferences.getString(host, "");
    }
    
    public void setAuthorization(String host, String authorization) {
        authPreferences.edit().putString(host, authorization).apply();
    }
    
    public String getToken(String host) {
        return tokenPreferences.getString(host, "");
    }
    
    public void setToken(String host, String token) {
        tokenPreferences.edit().putString(host, token).apply();
    }
    
    public String getUserName() {
        return privacyPreferences.getString("username", "");
    }
    
    public void setUserName(String userName) {
        privacyPreferences.edit().putString("username", userName).apply();
    }
    
    public String getPassword() {
        return privacyPreferences.getString("password", "");
    }
    
    public void setPassword(String password) {
        privacyPreferences.edit().putString("password", password).apply();
    }
    
    public String getCookie(String host) {
        return cookieManager.toSimpleString(host);
    }
    
    public Context getContext() {
        return context;
    }
    
    @NonNull
    @Override
    public String toString() {
        return authPreferences.toString();
    }
    
}

File: app/src/main/java/com/sysu/edu/api/AuthorizationJar.kt
package com.sysu.edu.api

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

class AuthorizationJar(val context: Context) {
	private val authPreferences: SharedPreferences = context.getSharedPreferences("authorization", Context.MODE_PRIVATE)
	private val tokenPreferences: SharedPreferences = context.getSharedPreferences("token", Context.MODE_PRIVATE)
	private val cookieManager: CookieManager = CookieManager(context)
	fun getAuthorization(host: String?): String {
		return authPreferences.getString(host, "") ?: ""
	}
	
	fun setAuthorization(host: String?, authorization: String?) {
		authPreferences.edit { putString(host, authorization) }
	}
	
	fun getToken(host: String?): String {
		return tokenPreferences.getString(host, "") ?: ""
	}
	
	fun setToken(host: String?, token: String?) {
		tokenPreferences.edit { putString(host, token) }
	}
	
	fun getCookie(host: String?): String {
		return cookieManager.toSimpleString(host)
	}
}
